tests/test_settings.py

```diff
@@ -32,73 +32,68 @@ def validate_int_list(val) -> bool:
     return True
 
 
-settings = SettingsManager()
-settings.register(option, default_val, description, validate_bool)
+@pytest.fixture
+def settings() -> SettingsManager:
+    settings = SettingsManager()
+    settings.register(option, default_val, description, validate_bool)
+    settings.register(option_2, default_val_2, description_2, validate_bool)
+    settings.register(option_3, default_val_3, description_3, validate_int_list, type_3)
+    return settings
 
-settings.register(option_2, default_val_2, description_2, validate_bool)
 
-settings.register(
-    option_3,
-    default_val_3,
-    description_3,
-    validate_int_list,
-    type_3,
-)
-
-
-def test_register_option_default():
+def test_register_option_default(settings: SettingsManager):
     assert getattr(settings, option) == default_val
     assert description in settings.describe(option)
 
 
-def test_register_with_env(monkeypatch):
-    with monkeypatch.context() as mp:
-        option_env = "test_var_env"
-        default_val_env = False
-        description_env = "My doc string env!"
-        option_env_var = "ANNDATA_" + option_env.upper()
-        mp.setenv(option_env_var, "1")
-
-        settings.register(
-            option_env,
-            default_val_env,
-            description_env,
-            validate_bool,
-            get_from_env=check_and_get_bool,
-        )
-
-        assert settings.test_var_env
-
-
-def test_register_with_env_enum(monkeypatch):
-    with monkeypatch.context() as mp:
-        option_env = "test_var_env"
-        default_val_env = False
-        description_env = "My doc string env!"
-        option_env_var = "ANNDATA_" + option_env.upper()
-        mp.setenv(option_env_var, "b")
-
-        class TestEnum(Enum):
-            a = False
-            b = True
+def test_register_with_env(settings: SettingsManager, monkeypatch: pytest.MonkeyPatch):
```

Comment:
Do you know if `monkeypatch` is correctly scoped to remove the patch once the test is torn down (i.e., no need for context)?  I did this to be safe because I couldn't find anything definite, although I suspect that it is correctly scoped (I just found this blog post: https://holgerkrekel.net/2009/03/03/monkeypatching-in-unit-tests-done-right/ but couldn't find anything in the docs, so I'm sure it's fine).  This is much cleaner